prometheus_client/metrics.py

```diff
@@ -203,6 +203,41 @@ def remove(self, *labelvalues: Any) -> None:
             if labelvalues in self._metrics:
                 del self._metrics[labelvalues]
 
+    def remove_matching(self, partial: dict[str, str]) -> int:
+        if 'prometheus_multiproc_dir' in os.environ or 'PROMETHEUS_MULTIPROC_DIR' in os.environ:
+            warnings.warn(
+                "Removal of labels has not been implemented in  multi-process mode yet.",
+                UserWarning
+            )
+
+        if not self._labelnames:
+            raise ValueError('No label names were set when constructing %s' % self)
+        
+        if not isinstance(partial, dict):
+            raise TypeError("partial must be a dict of {label_name: label_value}")
+    
+        if not partial:
+            return 0
+    
+        invalid = [k for k in partial.keys() if k not in self._labelnames]
+        if invalid:
+            raise ValueError(
+                'Unknown label names: %s; expected %s' % (invalid, self._labelnames)
+            )
+        
+        pos_filter = {self._labelnames.index(k): str(v) for k, v in partial.items()}
+
+        deleted = 0
```

Comment:
What's the reasoning for returning the number of deleted items? We don't do that in remove today, but if there is a good use case I am not opposed.